consensus/polybft/state_sync_manager.go

```diff
@@ -0,0 +1,526 @@
+package polybft
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"path"
+	"sync"
+
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/bitmap"
+	polybftProto "github.com/0xPolygon/polygon-edge/consensus/polybft/proto"
+	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
+	"github.com/0xPolygon/polygon-edge/tracker"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/umbracle/ethgo"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	// number of stateSyncEvents to be processed before a commitment message can be created and gossiped
+	stateSyncCommitmentSize = 10
+)
+
+// StateSyncManager is an interface that defines functions for state sync workflow
+type StateSyncManager interface {
```

Comment:
I do not see the use case to do this abstraction here. The only reason would be to put it inside `consensus_runtime` but even that I think this is lightweight enough to not mock it and use it directly.

Reply:
I agree, it's really lightweight to mock it. This interface is added only because of the `nil` implementation of `stateSyncManager`, which we discuss in comment: https://github.com/0xPolygon/polygon-edge/pull/1067#discussion_r1058790320